### DISCUSSION TITLE
片方のキーボードしか反応しない

### DISCUSSION THREAD

**Discussion by t4kamura** [Yowkees/keyball#701] [Q&A]
この度はお世話になります。
片方のキーボードしか反応しない不具合に悩んでおり、色々と試してはみたのですが解決せず、ご助言をいただけるとありがたいです。

# 動作環境

Keyball44
MacOS 15.1.1

# 事象

- PCと接続していない側のキーボードが一切反応しない
  - 例えば、右側をPCと接続して左側とTRRSケーブルで繋いだ場合、右しか反応しない
    - 左側のProMicroの赤いLEDは常に点灯する
    - 左側のディスプレイは映るときとそうでないときがある
  - 逆の場合でも同様にPCと接続していない方が一切反応しない

# 試したこと

テスターを用いて以下を確認しました

- VCC電圧を測定して3Vであること
- 左右のDATAの疎通が問題ないこと
- 4PソケットとSDA・SCLポートの疎通が問題ないこと

また[こちら](https://github.com/Yowkees/keyball/discussions/578)を参考にProMicroの電圧を5Vに変更しましたが、事象は変わらずでした。

他にも以下を試しております

- 左右ごとにファームウェアを書き直すこと
  - ファームウェアは[こちら](https://remap-keys.app/catalog/tAJ9Htme4oNabUkx4832/firmware)から書き込みました
  - `Keyball44_via_v1.3.2` と `Keyball44_test_v1.3.2` の両方を試しました
- TRRSケーブルの交換
  - 同じ種類のTRRSケーブルを新しく購入し交換してみましたが、変わらずでした

# 画像
![IMG_2512](https://github.com/user-attachments/assets/5c496130-fe88-4a9f-b341-bb90c799a42c)
![IMG_2511](https://github.com/user-attachments/assets/7b235351-f1fb-4a2e-9dc8-c5803f5f9387)
![IMG_2510](https://github.com/user-attachments/assets/3c503f14-a566-4074-9396-122909146d7e)

**Comment by Yowkees** [comment]
ご連絡ありがとうございます。

左右通信が不安定ということで、TRSジャックの半田付けを見直してみてください。
下記画像の緑丸の半田付けがちょっと怪しいかと思います。
半田がミドル基板の銀パッドに馴染んで、山型になるように新鮮な半田を追加してみてください。
![TRSfix](https://github.com/user-attachments/assets/0a0ff7d3-f6a4-4b37-841b-1f1e62c82e5d)

改善しない場合は、TRSジャックのその他の端子やProMicroの半田付けで隣とくっついていたり、ボール状になって銀パッドに馴染んでいないものを見つけて修正してみていただければ幸いです。

以上、お手数ですがご確認お願いいたします。

**Comment by t4kamura** [comment]
@Yowkees 
ありがとうございます。
色々と試したのですが改善せず、最終的に右手側のProMicroを新品のものに取り替えることで解決しました。
具体的な原因を特定できず申し訳ないのですが、解決とさせていただければと思います。
